interceptors/ratelimit/examples_test.go

```diff
@@ -2,32 +2,59 @@ package ratelimit_test
 
 import (
 	"context"
+	"fmt"
 
+	middleware "github.com/grpc-ecosystem/go-grpc-middleware/v2"
+	grpc_ratelimit "github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/ratelimit"
+	"github.com/juju/ratelimit"
 	"google.golang.org/grpc"
+)
 
-	middleware "github.com/grpc-ecosystem/go-grpc-middleware/v2"
-	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/ratelimit"
+// Rationale:
+// we use in the rate-limiter interceptor a "tocken-bucket" algorithm
+// check the article for more info
+// https://de.wikipedia.org/wiki/Token-Bucket-Algorithmus
+// alternatively you could use a leaky-bucket or other algorithms
+
+// here hard-coded for simplicity sake, but in your application you should make this configurable
+const (
+	// add 5 token per seconds
+	rate = 5
+	// capacity of bucket. allow only 40 requests
+	tokenCapacity = 40
 )
 
-// alwaysPassLimiter is an example limiter which implements Limiter interface.
-// It does not limit any request because Limit function always returns false.
-type alwaysPassLimiter struct{}
+type rateLimiterInterceptor struct {
+	tokenBucket *ratelimit.Bucket
```

Comment:
again I think this should land in provider similar to Prometheus and specific logging libs